### PR TITLE
Feature lens flare

### PR DESCRIPTION
**Pull request by DavidMansolino**
Linked to omichel/webots#4242

Link: https://www.cyberbotics.com/doc/reference/lensflare?version=feature-lens-flare

### CODE REVIEW

**Comment by DavidMansolino** [comment]
:+1:

File: reference/camera.md [hunk 1]
```diff
@@ -102,6 +103,9 @@ no focus is available on the camera device.
 with a controllable zoom system. If this field is set to NULL, then no zoom is
 available on the camera device.
 
+- The `lensFlare` field may contain a [LensFlare](lensflare.md) node to add lens flare
+in the camera image (if any light casts flares).
```
Comment (omichel) [inline reference/camera.md]: I would say "... to add a lens flare effect to the camera image..."

File: reference/lensflare.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+### Description
+
+%figure "Lens flare effect"
+
+![lens_flare.png](images/lens_flare.png)
+
+%end
+
+The [LensFlare](#lensflare) node can be added in the `lensFlare` field of any [Camera](camera.md) and [Viewpoint](viewpoint.md) to simulate lens flare. Lens flare effect is due to the light been scattered in lens systems through generally unwanted image formation mechanisms, such as internal reflection and scattering from material inhomogeneities in the lens. To simulate lens flare a variable number of images called `halos` and `bursts` are added along a line from the light position to a point in front of the [Camera](camera.md) or [Viewpoint](viewpoint.md).
```
Comment (omichel) [inline reference/lensflare.md]: and &rarr; or

Comment (omichel) [inline reference/lensflare.md]: Lens flare effect &rarr; The lens flare effect
been &rarr; being
reflection &rarr; reflections
To simulate lens flare &rarr; To simulate a lens flare,

File: reference/lensflare.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+### Field Summary
+
+- The `occlusion` field specifies whether or not obstacles between the [Camera](camera.md) or [Viewpoint](viewpoint.md) and the light should remove the lens flare effect. By default, the `occlusion` field is disabled because the occlusion detection is computationally expensive and should be avoided whenever possible.
```
Comment (omichel) [inline reference/lensflare.md]: remove &rarr; disable
disabled &rarr; set to FALSE
avoided whenever possible. &rarr; avoided when not needed.

File: reference/lensflare.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+- The `transparency` field specifies the transparency of the flare images.
```
Comment (omichel) [inline reference/lensflare.md]: the transparency of the flare images. &rarr; the transparency level of the flare images.
We should explain the range here and also the meaning. Is 1 fully transparent and 1 fully opaque?

File: reference/lensflare.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+- The `haloSizesAndPositions` field specifies the size and position along the line mentioned previously of each of the halo images. The first component defines the reference size of the halo image in meters (the reference size is the size of the halo image considering the light is at a distance of 1 meter, if the light is closer or farther, the image is scaled accordingly). The second component defines the position along the line, 0.0 means at light position and 1.0 means in front of the [Camera](camera.md)/[Viewpoint](viewpoint.md) (you are allowed to set values outside of this range, but it is recommended to remain in this range to have realistic effects). Furthermore the number of elements defines the number of halo images.
```
Comment (omichel) [inline reference/lensflare.md]: each of the halo &rarr; each halo

Comment (omichel) [inline reference/lensflare.md]: [Camera](camera.md)/[Viewpoint](viewpoint.md) &rarr; [Camera](camera.md) or [Viewpoint](viewpoint.md)

Comment (omichel) [inline reference/lensflare.md]: I would remove the parenthesis for the sentence about the range and make it a normal sentence.

Comment (omichel) [inline reference/lensflare.md]: Edit: of each of the halo images. → for each halo image.

File: reference/lensflare.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+- The `burstSizesAndPositions` field specifies the size and position along the line mentioned previously of each of the burst images. The first component defines the reference size of the burst image in meters (the reference size is the size of the burst image considering the light is at a distance of 1 meter, if the light is closer or farther, the image is scaled accordingly). The second component defines the position along the line, 0.0 means at light position and 1.0 means in front of the [Camera](camera.md)/[Viewpoint](viewpoint.md) (you are allowed to set values outside of this range, but it is recommended to remain in this range to have realistic effects). Furthermore the number of elements defines the number of burst images.
```
Comment (omichel) [inline reference/lensflare.md]: of each of the burst images. → for each burst image.

Comment (omichel) [inline reference/lensflare.md]: Same comments apply to this paragraph as in the previous one.

File: reference/viewpoint.md [hunk 7]
```diff
@@ -49,3 +50,6 @@ adding the instance number to the `follow` field value using the format
 "`<name>:<number>`".
 
 The `followOrientation` field can be used to make the viewpoint follow also the orientation of an object (in addition to its position). If `followOrientation` is true, the viewpoint is rigidly attached to the followed object, like an embedded camera onboard a robot. The `follow` field should be set with a valid object name otherwise the `followOrientation` field has no effect.
+
+The `lensFlare` field may contain a [LensFlare](lensflare.md) node to add lens flare
```
Comment (omichel) [inline reference/viewpoint.md]: &rarr to add a lens flare effect

**Comment by DavidMansolino** [comment]
:+1:

**Comment by omichel** [comment]
:+1:

**Comment by DavidMansolino** [comment]
:+1:

**Comment by DavidMansolino** [comment]
:+1:

**Comment by fabienrohrer** [comment]
👍

**Pull request merged**